extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java

```diff
@@ -0,0 +1,518 @@
+                try (SqlResult ignored = sql.execute(queries.storeUpdate(), params)) {
+                }
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    @Override
+    public void storeAll(Map<K, GenericRecord> map) {
+        awaitInitFinished();
+
+        for (Entry<K, GenericRecord> entry : map.entrySet()) {
+            store(entry.getKey(), entry.getValue());
+        }
+    }
+
+    @Override
+    public void delete(K key) {
+        awaitInitFinished();
+
+        try (SqlResult ignored = sql.execute(queries.delete(), key)) {
+        }
+    }
+
+    @Override
+    public void deleteAll(Collection<K> keys) {
+        awaitInitFinished();
+
+        if (keys.isEmpty()) {
+            return;
+        }
+
+        try (SqlResult ignored = sql.execute(queries.deleteAll(keys.size()), keys.toArray())) {
+        }
+    }
+
+    /**
+     * Returns true if the instance where the MapStore is instantiated is master
+     */
+    private boolean isMaster() {
+        return nodeEngine().getClusterService().isMaster();
+    }
+
+    private void dropMapping(String mappingName) {
+        try (SqlResult ignored = sql.execute("DROP MAPPING \"" + mappingName + "\"")) {
+        }
+    }
+
+    /**
+     * Awaits successful initialization, if the initialization failed it throws an exception
+     */
+    private void awaitInitFinished() {
+        try {
+            boolean finished = initFinished.await(initTimeoutMillis, MILLISECONDS);
+            if (!finished || initFailure != null) {
+                throw new HazelcastException("MapStore init failed for map: " + mapName, initFailure);
+            }
+        } catch (InterruptedException e) {
+            throw new HazelcastException(e);
+        }
+    }
+
+    private static class GenericMapStoreProperties {
+
+        final String externalDataStoreRef;
+        final String tableName;
+        final String mappingType;
+        final String idColumn;
+        final Collection<String> columns;
+        final boolean idColumnInColumns;
+        final String compactTypeName;
+
+        GenericMapStoreProperties(Properties properties, String mapName) {
+            externalDataStoreRef = properties.getProperty(EXTERNAL_REF_ID_PROPERTY);
+            tableName = properties.getProperty(TABLE_NAME_PROPERTY, mapName);
+            this.mappingType = properties.getProperty(MAPPING_TYPE_PROPERTY);
+            idColumn = properties.getProperty(ID_COLUMN_PROPERTY, ID_COLUMN_DEFAULT);
+
+            String columnsProperty = properties.getProperty(COLUMNS_PROPERTY);
+            if (columnsProperty != null) {
+                List<String> columns = new ArrayList<>();
+                Collections.addAll(columns, columnsProperty.split(","));
+                this.columns = unmodifiableList(columns);
+            } else {
+                columns = Collections.emptyList();
+            }
+            idColumnInColumns = columns.isEmpty() || columns.contains(idColumn);
```

Comment:
Why `columns.isEmpty()` ?

Reply:
If you don't set the columns property all columns are mapped. So it considers the id column to be _present_.

E.g. columns id, name, age
if you don't set the property you get GenericRecord with all fields
if you set e.g. name,age you get GenericRecord with only those, remember that you still have the id as the key of the map
you can also select e.g. id and name..

Hope it's clear :-)